src/compass/utils/stitching/stitch_burst.py

```diff
@@ -0,0 +1,404 @@
+
+    Returns:
+    -------
+    xres, yres, epsg: (tuple, float)
+       Raster resolution along X and Y directions and epsg
+    '''
+    raster = isce3.io.Raster(filename)
+    return raster.dx, raster.dy, raster.get_epsg()
+
+
+def save_as_shapefile(polygon, epsg, outfile):
+    '''
+    Save polygon as an ESRI shapefile
+
+    Parameters:
+    ----------
+    polygon: shapely.geometry.Polygon
+       Shapely polygon identify burst boundary on the ground
+    epsg: int
+       EPSG code associate to 'polygon' coordinate system
+    outfile: str
+       File path to store create ESRI shapefile
+
+    '''
+    dest_srs = ogr.osr.SpatialReference()
+    dest_srs.ImportFromEPSG(int(epsg))
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    ds = driver.CreateDataSource(outfile)
+    layer = ds.CreateLayer('', None, ogr.wkbPolygon)
+
+    # Add attribute and create new feature
+    layer.CreateField(ogr.FieldDefn('id', ogr.OFTInteger))
+    defn = layer.GetLayerDefn()
+    feat = ogr.Feature(defn)
+    feat.SetField('id', 123)
+
+    # Make a geometry, from Shapely object
+    geom = ogr.CreateGeometryFromWkb(polygon.wkb)
+    geom.AssignSpatialReference(dest_srs)
+    feat.SetGeometry(geom)
+    layer.CreateFeature(feat)
+
+    # Clean up
+    feat = geom = None
+    ds = layer = feat = geom = None
+
+
+def intersect_polygons(polygons, epsgs, margin):
+    '''
+    Get the intersection of polygons stored in 'polygons'
+
+    Parameters:
+    -----------
+    polygons: list
+       List of shapely polygons to intersect
+    epsgs: list
+       List of EPSGs associate to 'polygons'
+
+    Returns:
+    -------
+    poly_int: shapely.Polygon
+       Result of polygon intersection
+    epsg_int: int
+       EPSG code associated to poly
+    '''
+    # Assert validity of inputs
+    assert (len(polygons) == len(epsgs))
+
+    # Initialize polygon and epsg of intersection
+    poly_int = polygons[0]
+    epsg_int = epsgs[0]
+
+    # Initialize coordinate transformation in case
+    # there are polygons with different epsgs
+    llh = ogr.osr.SpatialReference()
+    llh.ImportFromEPSG(epsg_int)
+    tgt = ogr.osr.SpatialReference()
+
+    for poly, epsg in zip(polygons, epsgs):
+        if epsg != epsg_int:
+            # Transform polygons in same coord system as epsg_int
+            tgt_x, tgt_y = [], []
+            x, y = poly.exterior.coords.xy
+            tgt.ImportFromEPSG(epsg)
+            trans = ogr.osr.CoordinateTransformation(llh, tgt)
+            for lx, ly in zip(x, y):
+                dummy_x, dummy_y, dummy_z = trans.Transform(lx, ly, 0)
+                tgt_x.append(dummy_x)
+                tgt_y.append(dummy_y)
+            poly = Polygon(list(zip(tgt_x, tgt_y)))
+        poly_int = poly.intersection(poly_int)
+
+    # To be conservative, apply some margin to the polygon (TO DO:
+    # check eps)
+    poly_int = poly_int.buffer(-margin)
+    return poly_int, epsg_int
+
+
+def get_stitching_dict(indir):
```

Comment:
I made my suggestion because [this block of code](https://github.com/opera-adt/COMPASS/blob/6d5f80247a0baf1b146673471543cb34dc7ac6e2/src/compass/utils/stitching/stitch_burst.py#L89-L95):
```python
    data_dict = get_stitching_dict(indir)

    # If stitching some bursts, prune dataframe to
    # contains only the burst IDs to stitch
    if burst_list is not None:
        data_dict = prune_dataframe(data_dict,
                                    'burst_id', burst_list)
```
By [filtering out unwanted bursts in `get_stitching_dict`](https://github.com/opera-adt/COMPASS/pull/14#discussion_r915167917), the call to `prune_dataframe` is no longer needed and other calls to `prune_dataframe` will work on **smaller** dataframes. I wager a smaller `dataframe` will be faster to filter. If `burst_list is None`, downstream behavior is not impacted since everything in `indir`will be in `data_dict`.

My perspective of filtering is of what's present in the code of this PR. What others besides the 2 here that you have in mind?